### PR TITLE
#900 request req next button

### LINKED ISSUE #900: Request requisition line edit modal Next Button
Need to implement the functionality for the request requisition 'next' button

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #900 

- Another easy one.. not sure why I thought to do these in separate PRs! Nice to have a tight focus though 😁

### CODE REVIEW

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/hooks.tsx [hunk 1]
```diff
@@ -79,3 +79,25 @@ export const useDraftRequisitionLine = (item: ItemWithStatsFragment | null) => {
...
+  const idx = lines.findIndex(l => l.item.id === currentItem?.id);
+  const next = lines[idx + 1];
+  if (!next) {
+    nextState.isDisabled = true;
+    return nextState;
+  }
+
+  nextState.next = next.item;
+
+  return nextState;
```
Comment (joshxg) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/hooks.tsx:102]: Yeah the duplication pains me. I'm not seeing an easy way to make this generic as the use cases seem to be slightly different, though, this is likely going to be used a lot so maybe it's worth the time to try and do so

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 2]
```diff
@@ -26,18 +26,29 @@ export const RequestLineEdit = ({
   mode,
   item,
 }: RequestLineEditProps) => {
-  const isDisabled = useIsRequestRequisitionDisabled();
+  const disabled = useIsRequestRequisitionDisabled();
   const { Modal } = useDialog({ onClose, isOpen });
   const [currentItem, setCurrentItem] = useBufferState(item);
   const { draft, isLoading, save, update } =
     useDraftRequisitionLine(currentItem);
+  const { next, isDisabled } = useNextRequestLine(currentItem);
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:34]: same comment as for response reqs: `{ next, hasNext }` and reverting the name of `disabled => isDisabled` would be clearer overall

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/hooks.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/hooks.tsx:102]: heh. yeah - very similar but tricky to extract

**Review by mark-prins — APPROVED** [review]
👍  nice!

**Pull request merged**